src/vsr/replica.zig

```diff
@@ -6706,14 +6706,36 @@ pub fn ReplicaType(
             return true;
         }
 
+        /// Primary must have no missing headers and no faulty prepares between op_repair_min
+        /// and self.op, to maintain the invariant that a replica can repair everything back
+        /// till op_repair_min
+        fn primary_journal_repaired(self: *Self) bool {
+            assert(self.status == .normal or self.status == .view_change);
+            assert(self.primary_index(self.view) == self.replica);
+            assert(self.view == self.log_view);
+
+            if (!self.valid_hash_chain_between(self.op_repair_min(), self.op)) return false;
```

Comment:
I am _slightly_ concerned with performance here, as this is O(journal_slot_count), but I think we are not calling this on any hot paths, so should be fine.